garage/tf/q_functions/discrete_mlp_q_function.py

```diff
@@ -2,58 +2,72 @@
+        hidden_w_init: Initializer function for the weight
+                    of intermediate dense layer(s).
+        hidden_b_init: Initializer function for the bias
+                    of intermediate dense layer(s).
+        output_nonlinearity: Activation function for
+                    output dense layer.
+        output_w_init: Initializer function for the weight
+                    of output dense layer(s).
+        output_b_init: Initializer function for the bias
+                    of output dense layer(s).
+        layer_normalization: Bool for using layer normalization or not.
     """
 
     def __init__(self,
                  env_spec,
+                 name="discrete_mlp_q_function",
                  hidden_sizes=(32, 32),
                  hidden_nonlinearity=tf.nn.relu,
+                 hidden_w_init=tf.contrib.layers.xavier_initializer,
+                 hidden_b_init=tf.zeros_initializer,
                  output_nonlinearity=None,
-                 layer_norm=False):
-        """
-        Initialize class with multiple attributes.
+                 output_w_init=tf.contrib.layers.xavier_initializer,
+                 output_b_init=tf.zeros_initializer,
+                 layer_normalization=False):
+        obs_dim = env_spec.observation_space.shape
+        action_dim = env_spec.action_space.flat_dim
 
-        Args:
-            env_spec: environment specification
-            hidden_sizes: A list of numbers of hidden units
-                for all hidden layers.
-            hidden_nonlinearity: An activation shared by all fc layers.
-            output_nonlinearity: An activation used by the output layer.
-            layer_norm: A bool to indicate whether to perform
-                layer normalization or not.
-        """
-        super(DiscreteMLPQFunction, self).__init__()
+        self.model = DiscreteMLPModel(
+            output_dim=action_dim,
+            name=name,
+            hidden_sizes=hidden_sizes,
+            hidden_nonlinearity=hidden_nonlinearity,
+            hidden_w_init=hidden_w_init,
+            hidden_b_init=hidden_b_init,
+            output_nonlinearity=output_nonlinearity,
+            output_w_init=output_w_init,
+            output_b_init=output_b_init,
+            layer_normalization=layer_normalization)
+
+        obs_ph = tf.placeholder(tf.float32, (None, ) + obs_dim, name="obs")
 
-        self._action_dim = env_spec.action_space.flat_dim
-        self._hidden_sizes = hidden_sizes
-        self._hidden_nonlinearity = hidden_nonlinearity
-        self._output_nonlinearity = output_nonlinearity
-        self._layer_norm = layer_norm
+        self.model.build(obs_ph)
 
     @overrides
-    def build_net(self, name, input):
+    def get_qval_sym(self, input_var, name):
         """
-        Set up q network based on class attributes.
+        Symbolic graph for q-network.
 
         Args:
+            input_var: Input tf.placeholder to the network.
```

Comment:
what is the input_var? is it state?

Reply:
Yes

Reply:
can you update the variable names/docstrings accordingly?